support/util/imagemetadata.go

```diff
@@ -39,32 +42,85 @@ type RegistryClientImageMetadataProvider struct{}
 // tag is referring to could have changed. Once a digest is obtained, the cache is checked so that
 // no further fetching occurs. Only if both cache lookups fail, the image metadata is fetched and
 // stored in the cache.
-func (*RegistryClientImageMetadataProvider) ImageMetadata(ctx context.Context, imageRef string, pullSecret []byte, imageContentSource []hyperv1.ImageContentSource) (*dockerv1client.DockerImageConfig, error) {
+func (r *RegistryClientImageMetadataProvider) ImageMetadata(ctx context.Context, imageRef string, pullSecret []byte) (*dockerv1client.DockerImageConfig, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	var repo distribution.Repository
+	var ref *reference.DockerImageReference
 
-	ref, err := reference.Parse(imageRef)
+	parsedImageRef, err := reference.Parse(imageRef)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
 	}
 
-	// Check if we should override the image for another disconneceted image
-	if len(imageContentSource) > 0 {
-		overridedImageRef, err := GetRegistryOverrides(ctx, ref, imageContentSource)
-		if err != nil {
-			return nil, fmt.Errorf("failed to check registry overrides for image reference %q: %w", imageRef, err)
+	// If the image reference contains a digest, immediately look it up in the cache
+	if parsedImageRef.ID != "" {
+		if imageConfigObject, exists := imageMetadataCache.Get(parsedImageRef.ID); exists {
+			return imageConfigObject.(*dockerv1client.DockerImageConfig), nil
 		}
+	}
 
-		if overridedImageRef != nil {
-			ref = *overridedImageRef
+	// Check if we should override the image for another disconnected image
+	if len(r.OpenShiftImageRegistryOverrides) > 0 {
```

Comment:
is there a need for this if? I think we could simplify by
```
		ref = &parsedImageRef
		repo, err = getRepository(ctx, *ref, pullSecret)
		if err != nil {
			return nil, err
		}
		for source, mirrors := range r.OpenShiftImageRegistryOverrides {
			for _, mirror := range mirrors {
...
```

Reply:
Yeah there's basically two paths to get the repos: 1) one where we have image registry overrides (i.e. ICSPs/IDMSs) to handle 2) where there are no ICSP/IDMS to handle. We can't combine them as you suggested since in the case of 1) we need to swap out the registry with the mirror source first. 

I did remove the `else if` though.

Reply:
Isn't this
```	if len(r.OpenShiftImageRegistryOverrides) == 0 {
		ref = &parsedImageRef
		repo, err = getRepository(ctx, *ref, pullSecret)
		if err != nil {
			return nil, err
		}
	}

	// Check if we should override the image for another disconnected image
	if len(r.OpenShiftImageRegistryOverrides) > 0 {
```
than just

```
	ref = &parsedImageRef
	repo, err = getRepository(ctx, *ref, pullSecret)
	if err != nil {
		return nil, err
	}

	// Check if we should override the image for another disconnected image
	for source, mirrors := range r.OpenShiftImageRegistryOverrides {
		for _, mirror := range mirrors {
```

Reply:
Cross posting from slack:
```
ref = &parsedImageRef 
repo, err = getRepository(ctx, *ref, pullSecret) 
if err != nil { 
   return nil, err 
}
```
^ without the if, this will fail on disconnected, but that is expected since we aren’t using the mirror reference. In disconnected, we need to try and get the repo info for the mirrored image instead of the source image. If we do this first without any respect to whether or not there are OpenShiftImageRegistryOverrides, this would fail the function prematurely in a disconnected env.

Basically OpenShiftImageRegistryOverrides is the litmus test if we are in a connected or disconnected env there are any ICSPs/IDMSs that would tell us if we need to swap registry info before trying to get the repo to get the image metadata. (edited) 

Alberto:
ah, I see

then you only need one if, no?
```
	// Non disconnected environments.
	if len(r.OpenShiftImageRegistryOverrides) == 0 {
		ref = &parsedImageRef
		repo, err = getRepository(ctx, *ref, pullSecret)
		if err != nil {
			return nil, err
		}
	}

	// Disconnected environments.
	// Check if we should override the image for another disconnected image
	for source, mirrors := range r.OpenShiftImageRegistryOverrides {
		for _, mirror := range mirrors {
			ref, err = GetRegistryOverrides(ctx, parsedImageRef, source, mirror)
			if err != nil {
```

Bryan:
Yeah I think it can be simplified to that. I’ll update it.